project/src/behaviors/GSTPursue.gd

```diff
@@ -0,0 +1,44 @@
+extends GSTSteeringBehavior
+class_name GSTPursue
+"""
+Calculates acceleration to take an agent to intersect with where a target agent will be.
+
+The `max_predict_time` variable represents how far ahead to calculate the intersection point.
+"""
+
+
+var target: GSTSteeringAgent
+var max_predict_time: float
+
+
+func _init(
+		agent: GSTSteeringAgent,
+		target: GSTSteeringAgent,
+		max_predict_time: = 1.0).(agent) -> void:
+	self.target = target
+	self.max_predict_time = max_predict_time
+
+
+func _calculate_steering(acceleration: GSTTargetAcceleration) -> GSTTargetAcceleration:
+	var target_position: = target.position
+	var distance_squared: = (target_position - agent.position).length_squared()
```

Comment:
I guess apart from the `Utility` function, you use `*_squared` so let's just stick with that in `Utility.clampv3()` as well, it's more expressive anyway